pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -613,6 +619,70 @@ func (r *ReconcileComplianceRemediation) verifyAndCompleteMC(obj *unstructured.U
 	return nil
 }
 
+// Process kubeletconfig remediation
+func (r *ReconcileComplianceRemediation) verifyAndCompleteKC(obj *unstructured.Unstructured, rem *compv1alpha1.ComplianceRemediation) error {
+	scan := &compv1alpha1.ComplianceScan{}
+	scanKey := types.NamespacedName{Name: rem.Labels[compv1alpha1.ComplianceScanLabel], Namespace: rem.Namespace}
+	if err := r.client.Get(context.TODO(), scanKey, scan); err != nil {
+		return fmt.Errorf("couldn't get scan for KC remediation: %w", err)
+	}
+	mcfgpools := &mcfgv1.MachineConfigPoolList{}
+	if err := r.client.List(context.TODO(), mcfgpools); err != nil {
+		return fmt.Errorf("couldn't list the pools for the remediation: %w", err)
+	}
+	// The scans contain a nodeSelector that ultimately must match a machineConfigPool. The only way we can
+	// ensure it does is by checking if it matches any MachineConfigPool's labels.
+	// See also: https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md
+	ok, pool := utils.AnyMcfgPoolLabelMatches(scan.Spec.NodeSelector, mcfgpools)
+	if !ok {
+		return common.NewNonRetriableCtrlError("not applying remediation that doesn't have a matching MachineconfigPool. Scan: %s", scan.Name)
+	}
+
+	// We need to check if there is exsisting kubeletConfigMC present for selected MCP
+	hasCustomKC, kubeletMCName, err := utils.IsMcfgPoolUsingKC(pool)
+
+	if err != nil {
+		return err
+	}
+	// we need to patch the remediation if there is already a CustomKC present
+	if hasCustomKC {
+
+		kubletMC := &mcfgv1.MachineConfig{}
+		kMCKey := types.NamespacedName{Name: kubeletMCName}
+
+		if err := r.client.Get(context.TODO(), kMCKey, kubletMC); err != nil {
+			return fmt.Errorf("couldn't get current generated KubeletConfig MC: %w", err)
+		}
+		// WE need to get name of original kubelet config that used to generate this kubeletconfig machine config
+		if len(kubletMC.GetOwnerReferences()) == 0 {
+			return fmt.Errorf("the generated kubelet machine config does not have a owner")
+		}
+		kubletName := kubletMC.GetOwnerReferences()[0].Name
+
+		kubeletConfig := &mcfgv1.KubeletConfig{}
+		kcKey := types.NamespacedName{Name: kubletName, Namespace: rem.Namespace}
+		if err := r.client.Get(context.TODO(), kcKey, kubeletConfig); err != nil {
+			return fmt.Errorf("couldn't get current KubeletConfig: %w", err)
+		}
+		// Set kublet config name
+		obj.SetName(kubeletConfig.GetName())
+		obj.SetLabels(kubeletConfig.GetLabels())
+		return nil
+	}
+
+	// We will need to create a kubelet config if there is no custom KC
+	kubletName := "compliance-operator-kubelet"
+
+	// Set kublet config name
+	obj.SetName(kubletName)
+
+	// Set MachineConfigSelector
+	NodeSelector := []string{"spec", "machineConfigPoolSelector", "matchLabels"}
```

Comment:
```
type KubeletConfigSpec struct {
	MachineConfigPoolSelector *metav1.LabelSelector `json:"machineConfigPoolSelector,omitempty"`
	KubeletConfig             *runtime.RawExtension `json:"kubeletConfig,omitempty"`
}
```
I think we have to use MachineConfigPoolSelector insteade of node-selector here @JAORMX We will match the pool with `node-selector` first, then copy pool's `MachineConfigPoolSelector` to `KubeletConfig`